posts/2024-07-28-tree-sitter.markdown

```diff
@@ -0,0 +1,171 @@
+installed parsers.
+Of course, the effectiveness for a given parser depends on how well its queries are tested.
+
+When installing a parser, nvim-treesitter delegates to [`tree-sitter-cli`](https://github.com/tree-sitter/tree-sitter/blob/master/cli/README.md),
+which may in turn delegate to a C compiler.
+Some parsers need to be generated from a `grammar.js` file, which requires Node.js.
+
+With much of the heavy lifting having recently been moved from nvim-treesitter to
+`tree-sitter-cli`, installing parsers has become a lot less error-prone.
+
+However, setting up the correct toolchain [can still be a PITA on some platforms](https://github.com/nvim-treesitter/nvim-treesitter/wiki/Windows-support).
+
+And there's one pain point that cannot be solved by keeping track of compatible parsers
+in a lockfile...
+
+## Challenges with downstream plugin compatibility
+
+In March 2024, the tree-sitter-haskell parser underwent a complete rewrite.
+This significant update brought improvements
+but also broke compatibility with several downstream plugins, including:
+
+- [nvim-treesitter](https://github.com/nvim-treesitter/nvim-treesitter/pull/6580)
+- [neotest-haskell](https://github.com/mrcjkb/neotest-haskell/pull/162)
+- [haskell-snippets.nvim](https://github.com/mrcjkb/haskell-snippets.nvim/pull/27)
+- [vim-matchup](https://github.com/andymass/vim-matchup/pull/349)
+- [iswap.nvim](https://github.com/mizlan/iswap.nvim/pull/89)
+- [rainbow-delimiters.nvim](https://gitlab.com/HiPhish/rainbow-delimiters.nvim/-/commit/a0e715996c93187290c00d494d58e11d5f5e43ad)
+- [haskell-scope-highlighting.nvim](https://github.com/kiyoon/haskell-scope-highlighting.nvim/pull/3)
+- [nvim-treesitter-context](https://github.com/nvim-treesitter/nvim-treesitter-context/pull/447)
+- [nvim-treesitter-textobjects](https://github.com/nvim-treesitter/nvim-treesitter-textobjects/pull/613)
+- [tailwind-sorter.nvim](https://github.com/laytan/tailwind-sorter.nvim/pull/104)
+
+Synchronizing updates across all these plugins to ensure a seamless transition for users
+is nearly impossible, especially given the limited number of maintainers for these queries.
+
+### Immediate impact on users
+
+Until all downstream plugins have been updated, affected users are left with two primary options:
+
+- **Pinning nvim-treesitter:**
+  Users can pin nvim-treesitter to a version or revision that installs the old version of the parser.
+  Unfortunately, this workaround means they cannot benefit from any other parser updates or fixes.
+  Consequently, users may have to pin plugins that depend on other parsers, delaying overall ecosystem improvements.
+- **Disabling plugins:**
+  This approach ensures users can still receive updates for other parts of their setup
+  but at the cost of losing functionality provided by the disabled plugins.
+
+Wouldn't it be neat if you could pin parsers individually?
+
+## Enter rocks.nvim
+
+By now, it's a well-known fact that we (the nvim-neorocks team) have been using
```

Comment:
`By now, it's a well-known`
That seems a bit like a stretch considering the download stats on luarocks.org . Maybe reference the reddit announcements instead.